licenseplate_project/licenseplate_report.py

```diff
@@ -0,0 +1,178 @@
+        ms_time = (calendar.timegm(time.strptime(human, '%Y-%m-%d~%H:%M:%S')) * 1000) + 25200000
+        return ms_time
+
+    def namesCamera(self):
+        self.args.cameraName = self.args.cameraName.replace(", ", ",")
+        self.camera_list = self.args.cameraName.split(",")
+
+    def recentVehicle(self):
+        # url of the api
+        endpoint = self.api_url + "/api/vehicle/getRecentVehicleEvents"
+        #checks to see if the user put in a start and end time if not it defaults to one hour before current
+        if self.args.startTime:
+            start = self.milliseconds_time(self.args.startTime)
+        else:
+            start = int(round((time.time() - 1800) * 1000))
+        if self.args.endTime:
+            end = self.milliseconds_time(self.args.endTime)
+        else:
+            end = int(round(time.time() * 1000))
+        # any parameters
+        payload = {
+            'deviceUuids': self.cameraUuids,
+            "filterTypes": [self.args.filter],
+            "endTimeMs": end,
+            "startTimeMs": start
+        }
+        resp = self.api_sess.post(endpoint, json=payload,
+        verify=False)
+        content = resp.content
+        data = json.loads(content)
+        organized = json.dumps(resp.json(), indent=2, sort_keys=True)
+        return data
+
+    #adds the name, timestamp, camera name, and sighting number to a csv
+    def csv_add(self, value, data_camera):
+        timestamp = self.human_time(value['eventTimestamp'])
+        self.csv_data.append([])
+        if value['name'] == None:
+            self.name = 'Unidentified'
+        else:
+            self.name = value['name']
+        self.csv_data[self.count].append(self.name)
+        self.csv_data[self.count].append(value['vehicleLicensePlate'])
+        self.csv_data[self.count].append(timestamp)
+        camera = self.camera_name(value['deviceUuid'], data_camera)
+        self.csv_data[self.count].append(camera)
+        self.thumbnail = self.mediaBaseURL + value['thumbnailS3Key']
+        self.saving_img()
+        self.csv_data[self.count].append(self.fileName)
+        with open(self.args.report + '/' + self.args.csv + '.csv', 'w', newline = '') as f:
+            writer = csv.writer(f)     # create the csv writer
+            writer.writerow(self.header)    # write the header
+            writer.writerows(self.csv_data) # write the data
+
+
+    def execute(self):
+        self.count = 0
+        self.mediaBaseURL = "https://media.rhombussystems.com/media/faces?s3ObjectKey="
+        self.header = ['Vehicle', 'LicensePlate', 'Date', 'Camera', 'Image File Name']
+        self.csv_data = []
+        data_camera = self.camera_data()
+        self.namesCamera()
+        self.camera_uuid(data_camera)
+        data_recentVehicle = self.recentVehicle()
+        #gets a path and makes a directory file to the path
+        path = os.getcwd()
+        os.mkdir(path + '/' + self.args.report)
```

Comment:
This errors out if the directory already exists.  This needs to be changed in any python script that calls mkdir to either ignore the error or change the call.